system/doc/reference_manual/expressions.xml

```diff
@@ -1985,29 +1985,31 @@ KeyPattern := ValuePattern &lt;- MapExpression</pre>
     marker="stdlib:maps#iterator/2">maps:iterator/2</seemfa>.</p>
 
     <p>A <em>filter</em> is an expression that evaluates to
-    <c>true</c> or <c>false</c>, or a
-    <seeguide marker="#guard_expressions">guard expression</seeguide>.
-    If the filter is not a guard expression and evaluates
-    to a non-Boolean value <c>Val</c>, an exception
-    <c>{bad_filter, Val}</c> is triggered at runtime.</p>
+    <c>true</c> or <c>false</c>.</p>
 
     <p>The variables in the generator patterns shadow previously bound variables,
     including variables bound in a previous generator pattern.</p>
 
-    <p>A list comprehension returns a list, where the lists elements are the
+    <p>Variables bound in a generator expression is not visible outside the expression:</p>
+
+    <pre>
+1> <input>[{E,L} || E &lt;- L=[1,2,3]].</input>
+* 1:5: variable 'L' is unbound</pre>
+
+    <p>A <strong>list comprehension</strong> returns a list, where the lists elements are the
     result of evaluating <c>Expr</c> for each combination of generator
     elements for which all filters are true.</p>
 
-    <p>A bit string comprehension returns a bit string, which is
+    <p>A <strong>bit string comprehension</strong> returns a bit string, which is
     created by concatenating the results of evaluating <c>BitStringExpr</c> for
     each combination of bit string generator elements for which all
     filters are true.</p>
 
-    <p>A map comprehension returns a map, where the map elements are
-    the result of evaluating <c>KeyExpr</c> and <c>ValueExpr</c> for
-    each combination of generator elements for which all filters are
-    true. If the key expressions are not unique, the last occurrence
-    is stored in the map.</p>
+    <p>A <strong>map comprehension</strong> returns a map, where the
+    map elements are the result of evaluating <c>KeyExpr</c> and
```

Comment:
for consistency, if you previously write `where the list's elements`, then I would say here `where the map's elements`.
Essentially, pick one writing style and keep it consistent 😄